apps/ledger-live-mobile/src/newArch/contexts/InViewContext/index.tsx

```diff
@@ -0,0 +1,92 @@
+import React, {
+  createContext,
+  DependencyList,
+  type ReactNode,
+  type RefObject,
+  useCallback,
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from "react";
+import { Dimensions, type View } from "react-native";
+import { concatMap, from, interval } from "rxjs";
+import type { InViewOptions, InViewContext, InViewEntry, WatchedItem } from "./types";
+import { inViewStatus } from "./utils";
+
+const InViewContext = createContext<InViewContext>({});
+
+export function useInViewContext(
+  target: RefObject<View>,
+  onInViewUpdate: (entry: InViewEntry) => void,
+  deps: DependencyList = [],
+) {
+  const { addWatchedItem, removeWatchedItem } = useContext(InViewContext);
+  const onInViewUpdateCb = useCallback(onInViewUpdate, deps); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
is the bypass really necessary ? We should avoid that

Reply:
I might be missing another posibility but here's how I see it:

The way I see not to bypass the `react-hooks/exhaustive-deps` would be to go:
From version 1
```ts
// Declaration
function useInViewContext(
  target: RefObject<View>,
  onInViewUpdate: (entry: InViewEntry) => void,
  deps: DependencyList = [],
)
...
// Usage
useInViewContext(target, ({ isInView }) => {
  if (isInView) logImpressionCard(id);
}, [id]);
```
To version 2
```ts
// Declaration
function useInViewContext(
  target: RefObject<View>,
  onInViewUpdate: (entry: InViewEntry) => void
)
...
// Usage
useInViewContext(target, useCallback(({ isInView }) => {
  if (isInView) logImpressionCard(id);
}, [id]));
```
The problem I have with version 2 is that the typing is that the type signature doesn't make it clear that `onInViewUpdate` is a dependency and that as a result writing this is problematic:
```ts
useInViewContext(target, ({ isInView }) => {
  if (isInView) logImpressionCard(id);
});
```
(because the internal `useEffect` will rerender on every re-render).

In react we are used to pass functions to hooks directly (without wrapping them in `useCallback`. That's why I think that here forwarding the dependencies to the hook is better to avoid mistakes. However it means that the variable used to forward the dependencies can't be statically analized by `react-hooks/exhaustive-deps` which makes the bypass necessary.

That said I could add `useInViewContext` to the config `rules."react-hooks/exhaustive-deps".additionalHooks`. So I'll have to add `logImpressionCard` to all the deps.

Reply:
Done: https://github.com/LedgerHQ/ledger-live/pull/8526/commits/d444580c06da41f1debd703a2ebc98e7eb97a226